lib/api/response/survey_details_response.dart

```diff
@@ -1,21 +1,24 @@
 import 'package:flutter_survey/api/response/decoder/response_decoder.dart';
+import 'package:flutter_survey/api/response/question_response.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:json_annotation/json_annotation.dart';
 
 part 'survey_details_response.g.dart';
 
 @JsonSerializable()
 class SurveyDetailsResponse {
-  String? id;
-  String? title;
-  String? description;
-  String? coverImageUrl;
+  final String? id;
+  final String? title;
+  final String? description;
+  final String? coverImageUrl;
+  final List<QuestionResponse>? questions;
```

Comment:
should we use `explicitToJson: true` for this `QuestionResponse` type?

Reply:
@Shayokh144 we need it to convert the model to json only, not the reserve direction.